Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +12,198 @@ public static class BalanceStats
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0) return "N/A.";
+
+            int currentTopBalance = 0;
+            List<String> listOfPeople = new List<String>();
+
+            foreach (var values in peopleAndBalances)
+            {
+                var array = values.Split(", ");
+
+                // Parse the last value of each person
+                bool succes = int.TryParse(array[^1], out int parsed);
+
+                if (succes && parsed > currentTopBalance)
+                {
+                    listOfPeople.Clear();
+                    listOfPeople.Add(array[0]);
+                    currentTopBalance = parsed;
+                }
+                else if (succes && parsed == currentTopBalance)
+                {
+                    listOfPeople.Add(array[0]);
+                }
+            }
+            // Get the list of people
+            string persons = CountPeople(listOfPeople);
+            if (listOfPeople.Count == 1)
+            {
+                return $"{persons} is the richest person. ¤{currentTopBalance}.";
+            }
+            return $"{persons} are the richest people. ¤{currentTopBalance}.";
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0) return "N/A.";
+
+            int currentHigh = 99999;
+            List<String> listOfPeople = new List<String>();
+
+            foreach (var values in peopleAndBalances)
+            {
+                var array = values.Split(", ");
+                bool succes = int.TryParse(array[^1], out int parsed);
+
+                if (succes && parsed < currentHigh)
+                {
+                    listOfPeople.Clear();
+                    listOfPeople.Add(array[0]);
+                    currentHigh = parsed;
+                }
+                else if (succes && parsed == currentHigh)
+                {
+                    listOfPeople.Add(array[0]);
+                }
+            }
+            // Get the list of people
+            string persons = CountPeople(listOfPeople);
+
+            // Get a negative amount back with the sign after the minus sign
+            string amount = "";
+            if (currentHigh < 0) amount = $"-¤{-currentHigh}";
+            else amount = $"¤{currentHigh}";
+
+
+            if (listOfPeople.Count == 1)
+            {
+                return $"{persons} has the least money. {amount}.";
+            }
+            return $"{persons} have the least money. {amount}.";
+        }
+
+        // Return the name of the people
+        public static string CountPeople(List<string> people)
```

Comment:
This function claims to 'Count people' but does not return a count.

Reply:
Renamed it to ReturnPeople (variable naming is difficult)